python/BioSimSpace/Sandpit/Exscientia/Parameters/_parameters.py

```diff
@@ -1111,3 +752,195 @@ def _validate(molecule=None, tolerance=None, max_distance=None,
     if property_map is not None:
         if not isinstance(property_map, dict):
             raise TypeError("'property_map' must be of type 'dict'")
+
+# Wrapper function to dynamically generate functions with a given name.
+# Here "name" refers to the name of a supported AMBER protein force field.
+import sys as _sys
+_namespace = _sys.modules[__name__]
+def _make_amber_protein_function(name):
+    def _function(molecule, tolerance=1.2, max_distance=_Length(6, "A"),
+            water_model=None, leap_commands=None, bonds=None,
+            work_dir=None, property_map={}):
+        """Parameterise a molecule using the named AMBER force field.
+
+           Parameters
+           ----------
+
+           forcefield : str
+               The name of the AMBER force field.
+
+           molecule : :class:`Molecule <BioSimSpace._SireWrappers.Molecule>`, str
+               The molecule to parameterise, either as a Molecule object or SMILES
+               string.
+
+           tolerance : float
+               The tolerance used when searching for disulphide bonds. Atoms will
+               be considered to be bonded if they are a distance of less than
+               tolerance times the sum of the equilibrium bond radii apart.
+
+           max_distance : :class:`Length <BioSimSpace.Types.Length>`
+               The maximum distance between atoms when searching for disulphide
+               bonds.
+
+           water_model : str
+               The water model used to parameterise any structural ions.
+               Run 'BioSimSpace.Solvent.waterModels()' to see the supported
+               water models. This is ignored if ions are not present.
+
+           leap_commands : [str]
+               An optional list of extra commands for the LEaP program. These
+               will be added after any default commands and can be used to, e.g.,
+               load additional parameter files. When this option is set, we can no
+               longer fall back on GROMACS's pdb2gmx.
+
+           bonds : ((class:`Atom <BioSimSpace._SireWrappers.Atom>`, class:`Atom <BioSimSpace._SireWrappers.Atom>`))
+               An optional tuple of atom pairs to specify additional atoms that
+               should be bonded. This is useful when the PDB CONECT record is
+               incomplete.
+
+           work_dir : str
+                The working directory for the process.
+
+           property_map : dict
+               A dictionary that maps system "properties" to their user defined
+               values. This allows the user to refer to properties with their
+               own naming scheme, e.g. { "charge" : "my-charge" }
+
+           Returns
+           -------
+
+           molecule : :class:`Molecule <BioSimSpace._SireWrappers.Molecule>`
+               The parameterised molecule.
+        """
+        return _parameterise_amber_protein(
```

Comment:
Here `name` is first (which I think makes sense, because you can easily call `functools.partial` on it), while in `_parametrise_openff` `name` comes second. Should these two functions have the same parameter ordering?

Reply:
I agree. It would be good to make both of these consistent.